Practical Example/Neil Hiddink/RWContent/RWContent/Controllers/ContentViewController.swift

```diff
@@ -0,0 +1,205 @@
+                self.content += content
+                DispatchQueue.main.async {
+                    self.content = content.sorted { (first, second) -> Bool in
+                        first.attributes.released_at < second.attributes.released_at
+                    }
+                    self.tableView.reloadData()
+                    activityIndicator.stopAnimating()
+                    container.removeFromSuperview()
+                }
+            }
+        }
+    }
+    
+    private func setupAndDisplay(activityIndicator: UIActivityIndicatorView, container: UIView) {
+        // Adapted from https://coderwall.com/p/su1t1a/ios-customized-activity-indicator-with-swift
+        container.frame = view.frame
+        container.center = view.center
+        container.backgroundColor = UIColor(white: 0.3, alpha: 0.3)
+        container.layer.cornerRadius = 10
+        
+        let loadingView: UIView = UIView()
+        loadingView.frame = CGRect(x: 0, y: 0, width: 80, height: 80)
+        loadingView.center = view.center
+        loadingView.backgroundColor = UIColor(white: 0.3, alpha: 0.0)
+        loadingView.clipsToBounds = true
+
+        activityIndicator.style = .large
+        activityIndicator.color = .systemGray6
+        activityIndicator.frame = loadingView.frame
+        activityIndicator.center = CGPoint(x: loadingView.frame.width / 2, y: loadingView.frame.height / 2)
+
+        loadingView.addSubview(activityIndicator)
+        container.addSubview(loadingView)
+        view.addSubview(container)
+        activityIndicator.startAnimating()
+    }
+}
+
+extension ContentViewController: UITableViewDelegate, UITableViewDataSource {
+    
+    // MARK: - Delegate
+    
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
+        let baseURL = "https://www.raywenderlich.com"
+        
+        guard let path = (content[indexPath.row].links["self"] as? String)?.dropFirst(41) else { fatalError() }
+        guard let url = URL(string: baseURL + path) else { fatalError() }
+            
+        let configuration = SFSafariViewController.Configuration()
+        configuration.entersReaderIfAvailable = false
+            
+        // https://www.hackingwithswift.com/read/32/3/how-to-use-sfsafariviewcontroller-to-browse-a-web-page
+        let safariVC = SFSafariViewController(url: url, configuration: configuration)
+        safariVC.modalPresentationStyle = .overFullScreen
+        safariVC.preferredControlTintColor = .gray
+        present(safariVC, animated: true)
+    }
+    
+    // MARK: - Data Source
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 180.0
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if let cell = tableView.dequeueReusableCell(withIdentifier: "RWContentTableViewCell") as? RWContentTableViewCell {
```

Comment:
Prefer using `guard let` in cases where there's a precondition for your whole function.